### PR TITLE
feat(hybrid-cloud): Add an organization region map table

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
This table maps an organization to a region key. This region key is then used with SENTRY_REGION_CONFIGS to retrieve region details.

This table is a workaround for an issue that prevents new columns in the Organization table: https://github.com/getsentry/sentry/issues/38821

Once that issue is fixed a new migration should merge this table and the Organizaiton table

### CODE REVIEW

**Review by maxiuyuan — APPROVED** [review]

File: src/sentry/migrations/0318_add_org_region_map_table.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not. Generally
+    # we don't want to run in a transaction here, since for long running operations like data
+    # back-fills this results in us locking an increasing number of rows until we finally commit.
+    atomic = False
+
+    dependencies = [
+        ("sentry", "0316_add_context_field_to_groupowner"),
+        ("sentry", "0317_add_organization_region_map"),
```
Comment (markstory) [inline src/sentry/migrations/0318_add_org_region_map_table.py:28]: Not sure this migration should be included as 317 as the actual schema changes.

Comment (AniketDas-Tekky) [reply src/sentry/migrations/0318_add_org_region_map_table.py:28]: I was getting a merge failure before this.

Comment (AniketDas-Tekky) [reply src/sentry/migrations/0318_add_org_region_map_table.py]: Fixed

**Review by markstory — APPROVED** [review]

**Comment by wedamija** [comment]
I feel like we'd be better off figuring out the specific problem rather than hacking around this with a new table

**Comment by markstory** [comment]
> I feel like we'd be better off figuring out the specific problem rather than hacking around this with a new table

Agreed, I'm going to fix the problem with the feature flag + migration generation issue, but have been short on time this week :cry:

**Pull request closed without merging**